hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyPurgeResponse.java

```diff
@@ -82,10 +87,15 @@ public void addToDBBatch(OMMetadataManager omMetadataManager,
     if (fromSnapshot != null) {
       OmSnapshotManager omSnapshotManager =
           ((OmMetadataManagerImpl) omMetadataManager).getOzoneManager().getOmSnapshotManager();
-
+      IOzoneManagerLock lock = omMetadataManager.getLock();
+      UUID fromSnapshotId = fromSnapshot.getSnapshotId();
+      OMLockDetails lockDetails = lock.acquireReadLock(SNAPSHOT_DB_CONTENT_LOCK, fromSnapshotId.toString());
```

Comment:
do we only use read lock on SNAPSHOT_DB_CONTENT_LOCK?

Reply:
```
Acquire write GC_LOCK
Snapshot is flushed


   compact if first snapshot(kskip Compaction)

   take a checkpoint on tmp path
   update keyTable, fileTable, directory()
   Acquire write SNAPSHOT_DB_CONTENT_LOCK
    truncate and ingest other tables(dumpToSst and ingestSstFile)
    delete if new prodPath if exists and mv tmp to production version snapshotDB path
    bump version snapshotLocalDataYaml -> new versionPath
    SNapshot cache lock on snapshotId(previosu rocksdb instance)
        delete of old version of snapshot path
    release snapshot cache lock on snapshotId
   releaseWrite content Lock
Release gc lock
```